compiler/src/dotty/tools/dotc/core/tasty/TastyPickler.scala

```diff
@@ -82,5 +82,5 @@ class TastyPickler(val rootCls: ClassSymbol) {
     all.bytes
   }
 
-  val treePkl: TreePickler = new TreePickler(this)
+  val treePkl: TreePickler = new TreePickler(this: @unchecked)
```

Comment:
I checked the code, it seems we can move the `treePkl` instance to usage site. And it would align with how we deal with `PositionPickler` and `CommentPickler`. Performance-wise, they are the same. 

https://github.com/lampepfl/dotty/blob/main/compiler/src/dotty/tools/dotc/transform/Pickler.scala#L67-L71

https://github.com/lampepfl/dotty/blob/main/compiler/src/dotty/tools/dotc/quoted/PickledQuotes.scala#L156-L157

Reply:
Great! Done